src/solver/closure.go

```diff
@@ -19,8 +23,50 @@ func IsSupportedFunction(name string) bool {
 	return ok
 }
 
-// Struct containing information about the function that called the closure.
+// ClosureCallerInfo containing information about the function that called the closure.
 type ClosureCallerInfo struct {
 	Name     string          // caller function name
 	ArgTypes []meta.TypesMap // types for each arg for call caller function
 }
+
+func GetClosureName(fun *ir.ClosureExpr, curFunction, curFile string) string {
```

Comment:
Can we make it consistent with shape type names generation?
Right now shape-related code is located in `linter` package (generateShapeName).
Maybe we can put closure-related code there as well and then move them both if we understand if there is anything common between them? We'll also generate anonymous classes at some point, so I don't want autogen type related code be spread around a lot.